cmd/kubeadm/app/apis/kubeadm/v1beta4/defaults.go

```diff
@@ -74,6 +74,7 @@ func SetDefaults_InitConfiguration(obj *InitConfiguration) {
 }
 
 // SetDefaults_ClusterConfiguration assigns default values for the ClusterConfiguration
+// +k8s:defaulter-gen=covers
```

Comment:
Even if we decide to apply `+k8s:defaulter-gen=covers` to the entire ClusterConfiguration, we'd better add the logic for setting default values for EnvVars.

Reply:
SGTM, this might be better to do.
@chendave could you please verify if it works, like this?

@SataQiu 

> // SetDefaults_EnvVar assigns default values for EnvVar.
> // +k8s:defaulter-gen=covers

would `+k8s:defaulter-gen=covers` still be required in this case?

Reply:
Yes, `+k8s:defaulter-gen=covers` is still required.

Reply:
+k8s:defaulter-gen=covers not required for `SetDefaults_EnvVar` but still required for `SetDefaults_ClusterConfiguration`. 

commit msg still applicable, so no need to update.

Reply:
Emm... Maybe there's some misunderstanding here. 
I mean, we can remove `+k8s:defaulter-gen=covers` from `SetDefaults_ClusterConfiguration`, and just add this tag for `SetDefaults_EnvVar`. This means that we might need an alias for `corev1.EnvVar` to take over the defaulter logic.

Reply:
the defaults will still be generated without the tag for `SetDefaults_ClusterConfiguration`, set the annotation for a substructure won't help iiuc, also, I tested and verified it, pls let me know if something is wrong.

Reply:
Draft Commit: https://github.com/kubernetes/kubernetes/commit/1081b49e8ffc679d8b6f94bc4c1959def54399e3

Reply:
thanks @SataQiu , that's change LGTM as well. If we are going to cherry-pick, I'd prefer to this one.

Reply:
> Draft Commit: [1081b49](https://github.com/kubernetes/kubernetes/commit/1081b49e8ffc679d8b6f94bc4c1959def54399e3)
> 
> Please have a look @chendave @neolit123

my vote: if this commit from @SataQiu works let's use it.
it's better not have the `+k8s:defaulter-gen=covers` for cluster configuration.

release note needs update to NONE regardless:
>> kubeadm: nested defaulters for `ClusterConfiguration` is not generated with the annotat

this is not a mechanism many users understand or care about.

Reply:
@SataQiu  would you mind push a review to supersede this one?

Reply:
I have no permission to push code into your branch.
You can cherry-pick it by:
```sh
git remote add SataQiu https://github.com/SataQiu/kubernetes.git
git pull --all
git cherry-pick cdbce96390c44d23fbb0ef7037276787fd5cff13
```

Reply:
I meant you can push a new PR instead, I am fine with the cherry-pick that change to here, give me some mins.